File: schemas/sdg_xp_bank_history.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from enums.enums import SDGType


class SDGXPBankHistorySchemaBase(BaseModel):
    history_id: int
    xp_bank_id: int
    sdg: SDGType
    increment: float
    reason: Optional[str]
    is_shown: Optional[bool] = False
    timestamp: datetime

    model_config = {
        "from_attributes": True
    }


class SDGXPBankHistorySchemaFull(SDGXPBankHistorySchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

# Not directly derived from models
# Todo: Generate TS type
class NoSDGXPBankHistorySchemaBase(BaseModel):
    message: str = "No bank history found for the user."
    increment: float = 0.0
    reason: Optional[str] = "No updates available."

File: frontend/types/sdgXpBankHistory.ts
import type { SDGType } from "./enums";

export interface SDGXPBankHistorySchemaBase {
  historyId: number;
  xpBankId: number;
  sdg: SDGType;
  increment: number;
  reason?: string | null;
  isShown?: boolean | null;
  timestamp: string;
}

export interface SDGXPBankHistorySchemaFull extends SDGXPBankHistorySchemaBase {
  createdAt: string;
  updatedAt: string;
}
